client/battle/BattleWindow.cpp

```diff
@@ -64,6 +64,40 @@ BattleWindow::BattleWindow(BattleInterface & owner):
 	
 	const JsonNode config(JsonPath::builtin("config/widgets/BattleWindow2.json"));
 	
+	auto useSpellIfPossible = [this](int slot){
+		std::string spellIdentifier = persistentStorage["quickSpell"][std::to_string(slot)].String();
+		SpellID id;
+		try
+		{
+			id = SpellID::decode(spellIdentifier);
+		}
+		catch(const IdentifierResolutionException& e)
+		{
+			return;
+		}
+		if(id.hasValue() && owner.getBattle()->battleGetMyHero() && id.toSpell()->canBeCast(owner.getBattle().get(), spells::Mode::HERO, owner.getBattle()->battleGetMyHero()))
+		{
+			owner.castThisSpell(id);
+		}
+	};
```

Comment:
Probably better to turn it into proper method of BattleWindow, similar to other shortcut handlers

Reply:
Done :)